pos_default_partner/models/pos_order.py

```diff
@@ -0,0 +1,15 @@
+# copyright 2020 Akretion David BEAL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+
+    @api.model
+    def create(self, vals):
```

Comment:
Thanks for your contribution, but on 13.0. There exists a hook that would allow to simplify the code and cast session just once.

You can find it here:

https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/models/pos_order.py#L320

On 12.0, the hook was not there.